runtime/common/src/auctions.rs

```diff
@@ -779,12 +788,18 @@ mod tests {
 				.unwrap_or_default()
 		}
 
-		fn lease_period() -> Self::LeasePeriod {
-			10
+		fn lease_period_length() -> (BlockNumber, BlockNumber) {
+			(10, 0)
 		}
 
-		fn lease_period_index() -> Self::LeasePeriod {
-			(System::block_number() / Self::lease_period()).into()
+		fn lease_period_index(b: BlockNumber) -> (Self::LeasePeriod, bool) {
+			let (lease_period_length, offset) = Self::lease_period_length();
+			let b = b.saturating_sub(offset);
```

Comment:
Do we want lease period 0 to be [0, offset+length)? It made more sense to me to have it be [offset, offset+length) and return an `Option<Self::LeasePeriod>` which is `None` if `now < offset`.

Reply:
let me look how this will affect the API and any error handling.